src/plugins/intel_cpu/src/emitters/snippets/x64/jit_brgemm_copy_b_emitter.cpp

```diff
@@ -100,13 +98,11 @@ void jit_brgemm_copy_b_emitter::init_brgemm_copy(std::unique_ptr<matmul::jit_brg
     brgCopyKernelConf.tr_b_dt_sz = DnnlExtensionUtils::sizeOfDataType(static_cast<dnnl::memory::data_type>(brgCopyKernelConf.src_dt));
     brgCopyKernelConf.req_wei_vnni_downconvert = false;
 
-    if (is_with_amx) {
-        brgCopyKernelConf.isa = avx512_core_amx;
-        brgCopyKernelConf.s8s8_compensation_required = false;
-    } else {
-        brgCopyKernelConf.isa = src_dt == dnnl_data_type_t::dnnl_bf16 ? avx512_core_bf16 : avx512_core_vnni;
-        brgCopyKernelConf.s8s8_compensation_required = src_dt == dnnl_data_type_t::dnnl_s8;
-    }
+
+    brgCopyKernelConf.isa = jit_brgemm_emitter::get_primitive_isa(src_dt, is_with_amx);
+    // Note: this assert can be removed when we support transposes through BrgemmCopyB
+    OV_CPU_JIT_EMITTER_ASSERT(isa_has_int8_vnni(brgCopyKernelConf.isa), "BrgemmCopyB emitter requires VNNI support");
+    brgCopyKernelConf.s8s8_compensation_required = src_dt == ov::element::i8 && !is_with_amx;
```

Comment:
Minor:
Could you please add `!avx_vnni_2` also here? As for [transformation](https://github.com/openvinotoolkit/openvino/pull/25495#discussion_r1680644324).

Reply:
Sure, let's do that.
Also I moved all these bregemm and isa-related checks to `jit_brgemm_emitter::get_brgemm_type`. Please take a look